scripting/src/main/java/org/infinispan/scripting/impl/DataTypedCache.java

```diff
@@ -69,22 +77,27 @@ public V get(Object key) {
 
    @Override
    public CacheSet<K> keySet() {
-      CacheStream<K> stream = getDelegate().keySet().stream().map(this::toDataType);
-      return new SetAsCacheSet<>(stream.collect(Collectors.toSet()));
+      CacheStream<K> stream = getDelegate().keySet().stream().map(
+            new ValueToTypedValueFunction<>(dataTypedCacheManager.dataType, dataTypedCacheManager.marshaller));
+      return new SetAsCacheSet<>(stream.collect(
```

Comment:
Indeed it will bring all contents over and it's not ideal, but I could not find an easy solution to avoid that :\. I know you have done a lot of work inside the keySet...etc impls to do that, but I couldn't easily extract it out for this situation. Do you have any ideas?

It would be great if we simply had methods to expose keyStream, valueStream, entryStream and leave those methods returning `Set` instances behind :\